llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -1757,6 +1815,7 @@ IndexCallsiteContextGraph::IndexCallsiteContextGraph(
           if (AN.MIBs.empty())
             continue;
           CallsWithMetadata.push_back({&AN});
+          CallToFunc[{&AN}] = FS;
```

Comment:
It was a little hard for me to tell why brace initialization was needed since the type of AN is not specified. Is this initializing a CallInfo from AllocInfo?

Reply:
The braces are needed here and for line 1852 to convert from either AllocInfo* or CallsiteInfo* to IndexCall. CallToFunc is a map from CallTy, which is a template parameter and for the index is of type IndexCall (which can hold either alloc or callsite summary pointers). Without the braces there is a compiler error.

Reply:
Sorry, I meant CallToFunc is a map from CallInfo, which contains a CallTy, that is a template parameter

Reply:
It compiles if I change this to `CallToFunc[IndexCall(&AN)] = FS` (and similar for line 1852). Would that be clearer?

Reply:
Sure, it is a little clearer. There are other places which should also be updated to make it consistent so I don't have a strong opinion.

Reply:
The other places like this were on the adjacent lines, so I went ahead and did a bit of refactoring to clean them all up.